jbpm-services/jbpm-kie-services/src/main/java/org/jbpm/kie/services/impl/bpmn2/ProcessDescriptor.java

```diff
@@ -108,7 +108,7 @@ public void resolveReusableSubProcessNames( Collection<Process> deploymentProces
 
 
     public ProcessAssetDesc getProcess() {
-        return process;
+        return process.copy();
```

Comment:
Is copy always required? Maybe make it default to copy and configurable to not if its not needed?

Reply:
yes, it must be copied always to avoid this kind of issues as we can't rely on checks if something is already using it or not.